### PR TITLE
feat: Custom attribute sidebar list UX improvements

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description

This PR includes UX improvements for the custom attributes list in the conversation and contact sidebar.
**UX changes**
Previously, we had to open the custom attributes dropdown, select the attribute, click edit, and save the values to view all the available attributes in the account. Now we are showing all the attributes from the account in the sidebar (both contact and conversation).

Fixes https://linear.app/chatwoot/issue/CW-3117/custom-attribute-list

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
**Loom video**
https://www.loom.com/share/cc3db00bb56a4a75b437ff207705f936?sid=4fddb195-6c43-4c0b-9bce-64188cff8b4d

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/dashboard/assets/scss/_layout.scss [hunk 1]
```diff
@@ -12,6 +12,7 @@ body {
     'Segoe UI',
     Roboto,
     'Helvetica Neue',
+    Tahoma,
```
Comment (pranavrajs) [inline app/javascript/dashboard/assets/scss/_layout.scss:15]: Missed in the Tailwind PR.

File: app/javascript/dashboard/routes/dashboard/conversation/ConversationInfo.vue [hunk 2]
```diff
@@ -1,152 +1,103 @@
...
+);
+const platformName = computed(() => {
+  const { browser } = props.conversationAttributes;
+  if (!browser) {
+    return '';
+  }
+  const { platform_name: platform, platform_version: platformVersion } =
+    browser;
+  return `${platform || ''} ${platformVersion || ''}`;
+});
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/conversation/ConversationInfo.vue:40]: Can we extract the `browser` and `platform` objects from before to avoid multiple accesses to nested properties.
```suggestion
const browser = props.conversationAttributes.browser || {};
const browserName = computed(() => {
  const { browser_name: browser = '', browser_version: browserVersion = '' } = browser;
  return `${browser} ${browserVersion}`;
});

const browserLanguage = computed(() =>
  getLanguageName(props.conversationAttributes.browser_language)
);

const platform = browser.platform || {};
const platformName = computed(() => {
  const { platform_name: platform = '', platform_version: platformVersion = '' } = platform;
  return `${platform} ${platformVersion}`;
});
```

File: app/javascript/dashboard/routes/dashboard/conversation/ConversationInfo.vue [hunk 3]
```diff
@@ -1,152 +1,103 @@
...
-      return availableAttributes.length % 2 === 0 ? 'even' : 'odd';
+    {
+      content: createdAtIp,
+      title: 'CONTACT_PANEL.IP_ADDRESS',
     },
-  },
-};
+  ];
+  return conversationAttributes.filter(attribute => !!attribute.content.value);
+});
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/conversation/ConversationInfo.vue:73]: Can we remove the unnecessary computations inside the computed property like this?

```
const staticElements = computed(() => [
  {
    content: initiatedAt,
    title: 'CONTACT_PANEL.INITIATED_AT',
  },
  {
    content: browserLanguage,
    title: 'CONTACT_PANEL.BROWSER_LANGUAGE',
  },
  {
    content: referer,
    title: 'CONTACT_PANEL.INITIATED_FROM',
    type: 'link',
  },
  {
    content: browserName,
    title: 'CONTACT_PANEL.BROWSER',
  },
  {
    content: platformName,
    title: 'CONTACT_PANEL.OS',
  },
  {
    content: createdAtIp,
    title: 'CONTACT_PANEL.IP_ADDRESS',
  },
].filter(attribute => !!attribute.content.value));
```

File: app/javascript/dashboard/routes/dashboard/conversation/ConversationInfo.vue [hunk 4]
```diff
@@ -1,152 +1,103 @@
...
+        :href="referer"
+        rel="noopener noreferrer nofollow"
+        target="_blank"
+        class="text-woot-400 dark:text-woot-600"
+      >
+        {{ referer }}
+      </a>
+    </ContactDetailsItem>
+    <CustomAttributes
+      :class="staticElements.length % 2 == 0 ? 'even' : 'odd'"
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/conversation/ConversationInfo.vue:97]:
```suggestion
      :class="staticElements.length % 2 === 0 ? 'even' : 'odd'"
```

**Comment by muhsin-k** [comment]
@iamsivin It seems like there are styles issue for conversation attributes.
CleanShot 2024-03-11 at 12 59 06@2x

**Comment by iamsivin** [comment]
> @iamsivin It seems like there are styles issue for conversation attributes. ![CleanShot 2024-03-11 at 12 59 06@2x]

@muhsin-k FIxed

**Review by scmmishra — APPROVED** [review]

File: app/javascript/dashboard/assets/scss/_layout.scss [hunk 5]
```diff
@@ -29,7 +30,7 @@ body {
   }
 }
 
-.banner + .app-wrapper {
+.banner+.app-wrapper {
```
Comment (scmmishra) [inline app/javascript/dashboard/assets/scss/_layout.scss:33]: Is it trivial to move these styles to tailwind classes directly? If it is let's do it in a separate PR

**Review by scmmishra — APPROVED** [review]

File: app/javascript/dashboard/assets/scss/_layout.scss (hunk 5 above)
Comment (iamsivin) [inline app/javascript/dashboard/assets/scss/_layout.scss:33]:
```suggestion
.banner + .app-wrapper {
```

Comment (iamsivin) [reply app/javascript/dashboard/assets/scss/_layout.scss]: Sure we can do it in a separate PR, for now I just removed this line from this PR

**Comment by iamsivin** [comment]
@pranavrajs  Could you please take a look and merge this PR?
